### PR TITLE
Masking workspace constants on frontend

### PR DESCRIPTION
**Pull request by AnantshreeChandola**
1. Backend Resolution of Workspace-Constants Values:
  All workspace-constants values are resolved entirely on the backend. This centralized handling ensures consistency and 
  security across all points of access.
2. Transmission of Encrypted Values:
 Except on the 'Mange workspace constants' page, values are not sent to the frontend. This is done to  
 ensure that the constants' values are not exposed during transmission or in client-side environments.
3. Permission Checks on API Access for workspace constants handling:
 The GET APIs for workspace constants include a step to resolve the user's permissions to create, update, or delete 
 workspace constants along with a query param 'decryptValue' to ensure that only authorized users can access the 
 decrypted values of the constants and only on the 'Manage workspace constants' page.
 4. Added logic to verify constants keys used in codehinter.

Resolves https://github.com/ToolJet/ToolJet/issues/9412

### CODE REVIEW

File: server/src/controllers/organization_constants.controller.ts [hunk 1]
```diff
@@ -31,24 +33,43 @@ export class OrganizationConstantController {
 
   @UseGuards(JwtAuthGuard)
   @Get()
-  async get(@User() user) {
-    const result = await this.organizationConstantsService.allEnvironmentConstants(user.organizationId);
+  async get(@User() user, @Req() req) {
```
Comment (gsmithun4) [inline server/src/controllers/organization_constants.controller.ts:36]: @Query('decryptValue') decryptValue

File: server/src/controllers/organization_constants.controller.ts [hunk 2]
```diff
@@ -31,24 +33,43 @@ export class OrganizationConstantController {
...
+        ability.can('deleteOrganizationConstant', OrganizationConstant));
+    const result = await this.organizationConstantsService.allEnvironmentConstants(user.organizationId, decrypt);
     return { constants: result };
   }
 
   @UseGuards(IsPublicGuard)
   @Get(':app_slug')
-  async getConstantsFromApp(@App() app) {
-    const result = await this.organizationConstantsService.allEnvironmentConstants(app.organizationId);
+  async getConstantsFromApp(@App() app, @User() user, @Req() req) {
```
Comment (gsmithun4) [inline server/src/controllers/organization_constants.controller.ts:49]: @query('decryptValue') decryptValue

File: server/src/controllers/organization_constants.controller.ts [hunk 3]
```diff
@@ -31,24 +33,43 @@ export class OrganizationConstantController {
...
+      (ability.can('createOrganizationConstant', OrganizationConstant) ||
+        ability.can('deleteOrganizationConstant', OrganizationConstant));
+    const result = await this.organizationConstantsService.allEnvironmentConstants(app.organizationId, decrypt);
     return { constants: result };
   }
 
   @UseGuards(JwtAuthGuard)
   @Get('/environment/:environmentId')
-  async getConstantsFromEnvironment(@User() user, @Param('environmentId') environmentId) {
+  async getConstantsFromEnvironment(@User() user, @Param('environmentId') environmentId, @Req() req) {
```
Comment (gsmithun4) [inline server/src/controllers/organization_constants.controller.ts:62]: @query('decryptValue') decryptValue

File: server/src/controllers/organization_constants.controller.ts [hunk 4]
```diff
@@ -31,24 +33,43 @@ export class OrganizationConstantController {
...
+    const result = await this.organizationConstantsService.allEnvironmentConstants(user.organizationId, decrypt);
     return { constants: result };
   }
 
   @UseGuards(IsPublicGuard)
   @Get(':app_slug')
-  async getConstantsFromApp(@App() app) {
-    const result = await this.organizationConstantsService.allEnvironmentConstants(app.organizationId);
+  async getConstantsFromApp(@App() app, @User() user, @Req() req) {
+    const ability = await this.organizationConstantsAbilityFactory.organizationConstantActions(user, null);
```
Comment (gsmithun4) [inline server/src/controllers/organization_constants.controller.ts:50]: This route is used for getting data for public apps, do we need `decryptValue` here?

File: frontend/src/Editor/Editor.jsx [hunk 5]
```diff
@@ -373,7 +375,7 @@ const EditorComponent = (props) => {
       const orgConstants = {};
       constants.map((constant) => {
         const constantValue = constant.values.find((value) => value.environmentName === 'production')['value'];
```
Comment (gsmithun4) [inline frontend/src/Editor/Editor.jsx:377]: `constantValue` is not used, can remove

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

**Review by shah21 — DISMISSED** [review]

**Review by gsmithun4 — APPROVED** [review]

**Pull request merged**